src/handler/BallotHandler.ts

```diff
@@ -47,4 +59,52 @@ export class BallotHandler {
 
     await ballotService.updateBallot(ballotID, running, question, options);
   }
+
+  public async getAdminStatus(
+    ballotID: string
+  ): Promise<VoteResult[]> {
+    if (!(await ballotService.checkIfBallotIDExists(ballotID)))
+      throw new NotFound("Ballot with given id does not exist");
+
+    const ballot = await ballotService.getBallot(ballotID);
+    const votes = await voteService.getVotes(ballotID);
+
+    const results: Record<string, number> = {} as never
+
+    // init vote counter
+    for (const voteOption of ballot.options) {
+      results[voteOption.identifier] = 0
+    }
+
+    // count votes
+    for (const vote of votes) {
+      results[vote.vote]++;
+    }
+
+    // fill records into array list
+    const out: VoteResult[] = [];
+    for (const result of Object.getOwnPropertyNames(results)) {
+      const voteOption = ballot.options.find(vo => result == vo.identifier)
+      const vr: VoteResult = {
+        questionIdentifier: voteOption.identifier,
+        questionLabel: voteOption.label,
+        amount: results[result]
+      }
+      out.push(vr);
+    }
+    return out;
+  }
+
+  public async getTotalVoteCount(
+    ballotID: string
+  ): Promise<TotalVoteCount> {
+    if (!(await ballotService.checkIfBallotIDExists(ballotID)))
+      throw new NotFound("Ballot with given id does not exist");
+
+    const votes = await voteService.getVotes(ballotID);
+
+    return {
+      count: votes.length
+    } as TotalVoteCount;
```

Comment:
Don' use type assertion here. It should generally be avoided in most cases.

Reply:
Also abstract more of this function into a service.
Handlers should optimally combine functions of services to handle conditional workflows like checking whether something exists.

Reply:
Resolved in [#54 Moved counting logic to voting service and refactored](https://github.com/LeoTuet/backend-we-want-to-sleep/pull/54/commits/d2c6e6ec6461eb9eec43029b972964306cbd4c12)